File: frontend/src/domain/model/Recipe.ts
/**
 * Domain model for a recipe summary (list views).
 */
export interface RecipeSummary {
  id: number;
  title: string;
  categoryName: string;
  viewCount: number;
}

/**
 * Domain model for a full recipe detail.
 */
export interface RecipeDetail {
  id: number;
  title: string;
  ingredients: string[];
  preparation: string;
  categoryId: number;
  categoryName: string;
  viewCount: number;
}

File: backend/src/main/java/be/lutske/leolegacy/domain/recipe/Recipe.java
package be.lutske.leolegacy.domain.recipe;

import be.lutske.leolegacy.domain.category.CategoryId;
import be.lutske.leolegacy.domain.category.CategoryName;
import be.lutske.leolegacy.domain.viewcount.ViewCount;
import java.time.Instant;

public class Recipe {
    private final RecipeTitle title;
    private final Ingredients ingredients;
    private final Preparation preparation;
    private ViewCount viewCount;

    public Recipe(RecipeTitle title, Ingredients ingredients, Preparation preparation, ViewCount viewCount) {
        this.title = title;
        this.ingredients = ingredients;
        this.preparation = preparation;
        this.viewCount = viewCount;
    }

    // existing getters and methods




    public RecipeTitle getTitle() {
        return title;
    }

    public Ingredients getIngredients() {
        return ingredients;
    }

    public Preparation getPreparation() {
        return preparation;
    }

    public CategoryId getCategoryId() {
        return categoryId;
    }

    public ViewCount getViewCount() {
        return viewCount;
    }

    public String getSource() {
        return source;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public String getImportMetadata() {
        return importMetadata;
    }

    public void incrementViewCount() {
        this.viewCount = new ViewCount(this.viewCount.getValue() + 1);
    }
}